palantirnet/st-bernards
# Switch from using files to media entities.

## JIRA Ticket(s)

- See ticket [MSSB-52: Create Media Bundles](https://palantir.atlassian.net/browse/MSSB-52).
- See ticket [MSSB-53: Setup core Media Library](https://palantir.atlassian.net/browse/MSSB-53).
- See ticket [MSSB-54: Update fields on CTs and Paragraph to use media/browser](https://palantir.atlassian.net/browse/MSSB-54).
- See ticket [MSSB-55: Migrate files/images to media bundles](https://palantir.atlassian.net/browse/MSSB-55).

## Description

Replace files with media entities. Update media modules, install and configure medial library, swap out existing file fields, migrate content into media entities and new fields. 

## Testing instructions

1. `composer install`
2. make sure you have a database from prod
3. `drush updb
drush cim`
4. you should now have all the media entities, libraries/browsers, and fields set up
5. prepare the migration 
`drush migrate:duplicate-file-detection media_news_images_step1
drush migrate:duplicate-file-detection media_location_images_step1
drush migrate:duplicate-file-detection media_any_slider_image_step1
drush migrate:duplicate-file-detection media_gallery_image_step1
drush migrate:duplicate-file-detection media_hero_icons_and_links_step1
drush migrate:duplicate-file-detection media_hero_image_step1
drush migrate:duplicate-file-detection media_inline_image_step1
drush migrate:duplicate-file-detection media_pdf_file_step1`
6. run the migrations `drush mim --group=media`
7. log in, you might need to unlock user 1 `drush user-unblock developer`
8. verify that media entities have been created and that they are being used in nodes and paragraphs.

## Additional Notes:

I did not see any existing videos or zip files so there are no migrations for them 


---

[Pull Request Guidelines](https://github.com/palantirnet/development_documentation/blob/master/guidelines/pr_review.md)
